src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespacebeforecasedefaultcolon/InputNoWhitespaceBeforeCaseDefaultColonEnumAndStrings.java

```diff
@@ -0,0 +1,81 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.whitespace.nowhitespacebeforecasedefaultcolon;
+
+/*
+ * Config: default
+ */
+class InputNoWhitespaceBeforeCaseDefaultColonEnumAndStrings {
+
+    enum Day {
+        MON,
+        TUE,
+        WED,
+        THU,
+        FRI,
+        SAT,
+        SUN,
+    }
+
+    String whatIsToday(Day day) {
+        return switch (day) {
+            case SAT, SUN -> "Weekend";
+            case MON, TUE, WED, THU, FRI -> "Working day";
+            default -> throw new IllegalArgumentException("Invalid day: " + day.name());
+        };
+    }
+
+    int assignement(Day day) {
```

Comment:
I wonder why spellcheck did not catch this.

Reply:
Resources are not checked by spell check, so this does not raise any violation.

https://github.com/checkstyle/checkstyle/blob/b15820a77a1cc27c3ad247602f4c89f442d22bdb/.ci/jsoref-spellchecker/exclude.pl#L11